lib/HLSL/DxilValidation.cpp

```diff
@@ -1658,10 +1658,12 @@ static void ValidateResourceDxilOp(CallInst *CI, DXIL::OpCode opcode,
   case DXIL::OpCode::CalculateLOD: {
     DxilInst_CalculateLOD lod(CI);
     Value *samplerHandle = lod.get_sampler();
-    if ((DXIL::CompareVersions(ValCtx.m_DxilMajor, ValCtx.m_DxilMinor, 1, 8) <
-         0) &&
-        GetSamplerKind(samplerHandle, ValCtx) != DXIL::SamplerKind::Default) {
-      ValCtx.EmitInstrError(CI, ValidationRule::InstrSamplerModeForLOD);
+    DXIL::SamplerKind samplerKind = GetSamplerKind(samplerHandle, ValCtx);
+    if (samplerKind != DXIL::SamplerKind::Default) {
+      // After SM68, Comparison is supported.
+      if (!ValCtx.DxilMod.GetShaderModel()->IsSM68Plus() ||
```

Comment:
As far as I can tell, this is currently equivalent to checking the dxil versions explicitly since they derive from the dxil modules shader model version. I know we've toyed with the idea of making the dxil version independent of shader model version, but this is hardly the only place in DxilValidation where this is done. 

My offhanded suggestion wasn't to use the shader model version, but to create a set of convenience functions for dxil version similar to these for the shader model version, but this is consistent too.